source/common/router/rds_impl.h

```diff
@@ -138,21 +122,62 @@ class RdsRouteConfigProviderImpl
   void registerInitTarget(Init::Manager& init_manager);
   void runInitializeCallbackIfAny();
 
-  Server::Configuration::FactoryContext& factory_context_;
   std::unique_ptr<Envoy::Config::Subscription<envoy::api::v2::RouteConfiguration>> subscription_;
-  ThreadLocal::SlotPtr tls_;
   std::string config_source_;
+  std::function<void()> initialize_callback_;
   const std::string route_config_name_;
-  absl::optional<LastConfigInfo> config_info_;
   Stats::ScopePtr scope_;
   RdsStats stats_;
-  std::function<void()> initialize_callback_;
   RouteConfigProviderManagerImpl& route_config_provider_manager_;
   const std::string manager_identifier_;
-  envoy::api::v2::RouteConfiguration route_config_proto_;
+  SystemTimeSource& time_source_;
   SystemTime last_updated_;
+  absl::optional<LastConfigInfo> config_info_;
+  envoy::api::v2::RouteConfiguration route_config_proto_;
+  std::unordered_set<RdsRouteConfigProviderImpl*> route_config_providers_;
+
+  friend class RouteConfigProviderManagerImpl;
+  friend class RdsRouteConfigProviderImpl;
```

Comment:
can we not use friend class?  define some getter functions.

Reply:
What is the point of define more getter functions? No other classes should access them via getter than friend classes listed here.